larg4/Core/larg4Main_module.cc

```diff
@@ -148,70 +150,67 @@ larg4::larg4Main::larg4Main(fhicl::ParameterSet const & p)
-	}
+  // If we're in "visualize specific events" mode (essentially only pause
+  // after given events), then extract the list of events we need to
+  // pause for. They are placed in a map because it is more efficient to
+  // determine whether a given entry is present in the map than a vector.
+  if (visSpecificEvents_) {
+    std::vector<int> eventsToDisplayVec =
+            p.get<vector<int>>("eventsToDisplay");
+    for (size_t i = 0; i < eventsToDisplayVec.size(); i++) {
+      eventsToDisplay_[eventsToDisplayVec[i]] = true;
+    }
+    // Would be nice to have error checking here, but for now, if you
+    // do something silly, it'll probably just crash.
+  }
 
   // We need all of the services to run @produces@ on the data they will store. We do this
   // by retrieving the holder services.
   art::ServiceHandle<ActionHolderService> actionHolder;
   art::ServiceHandle<DetectorHolderService> detectorHolder;
 
-    detectorHolder->initialize();
-    // Build the detectors' logical volumes
-    detectorHolder -> constructAllLVs();
-    // And running @callArtProduces@ on each
-    actionHolder -> callArtProduces(producesCollector());
-    detectorHolder -> callArtProduces(producesCollector());
-
-    //    ((artg4tk::SteppingActionBase*)&*pla)-> callArtProduces(this);
-    // ((artg4tk::EventActionBase*)&*pla) -> callArtProduces(this);
-    // ((artg4tk::TrackingActionBase*)&*pla) -> callArtProduces(this);
-  // Set up the random number engine.
-  // See the documentation in RandomNumberHeader.h for
-  // how this works. Note that @createEngine@ is a member function
-  // of our base class (actually, a couple of base classes deep!).
-  // Note that the name @G4Engine@ is special.
-  if (seed_ == -1) {
-	  // Construct seed from time and pid. (default behavior if
-	  // no seed is provided by the fcl file)
-	  // Note: According to Kevin Lynch, the two lines below are not portable.
-	  seed_ = time(0) + getpid();
-	  seed_ = ((seed_ & 0xFFFF0000) >> 16) | ((seed_ & 0x0000FFFF) << 16); //exchange upper and lower word
-	  seed_ = seed_ % 900000000; // ensure the seed is in the correct range for createEngine
+  detectorHolder->initialize();
+  // Build the detectors' logical volumes
+  detectorHolder -> constructAllLVs();
+  // And running @callArtProduces@ on each
+  actionHolder -> callArtProduces(producesCollector());
+  detectorHolder -> callArtProduces(producesCollector());
+
+  // ((artg4tk::SteppingActionBase*)&*pla)-> callArtProduces(this);
+  // ((artg4tk::EventActionBase*)&*pla) -> callArtProduces(this);
+  // ((artg4tk::TrackingActionBase*)&*pla) -> callArtProduces(this);
+
+  // -- Check for invalid seed value
+  if (seed_ > 900000000) {
+    //mf::LogError("SeedCheck") << "Bad seed provided, max seed value is 9E8.
+    throw cet::exception("largeant:BadSeedValue")
+          << "The provided largeant seed value: " << seed_
+          << " is invalid! Maximum seed value is 9E8.";
   }
```

Comment:
This check is not necessary--it is already done by `art` when `createEngine` is called.

Reply:
There was no error thrown when I set the seed to 1E9 in the fhicl file. 

```
%MSG-i NuRandomService:  larg4Main:largeant@Construction  24-Mar-2020 20:07:36 CDT ModuleConstruction
Seeding G4Engine engine "largeant:" with seed 1000000000.
   ...
Summary of seeds computed by the NuRandomService
Random policy: 'perEvent'
  algorithm version: EventTimestamp_v1
  Configured value          Last value   ModuleLabel.InstanceName
         1000000000              (same)   largeant [overridden]
```

Reply:
Which version of art were you using when running that job?

Reply:
`v3_05_00` I'm building off of larsoft suite v08_47_00

Reply:
Yikes!  I wouldn't have expected that.  The only way a value greater than 9E8 should be allowed is if you're using the `MixMaxRng` random number generator.  In that case, it's supported.  Are you by chance setting 
```
services.RandomNumberGenerator.defaultEngineKind: "MixMaxRng"
``` 
in your configuration file?

Reply:
No, this parameter isn't set in my fcl file. I dumped it and this is all there is for the NuRandomService at least.
```
     NuRandomService: {
        endOfJobSummary: true
        policy: "perEvent"
     }
```